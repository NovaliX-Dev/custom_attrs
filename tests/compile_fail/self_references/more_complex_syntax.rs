use custom_attrs::CustomAttrs;

#[derive(CustomAttrs)]
#[attr(pub a: usize)]
enum Enum {
    #[attr(a = #self.5)]
    Variant1(usize, u32),

    #[attr(a = #self.unknown)]
    Variant2 {
        #[allow(unused)]
        list: &[u32],

        #[allow(unused)]
        field2: usize,
    },
}

fn main() {
    let _a = Enum::Variant1(0, 1).get_a();
    let _b = Enum::Variant2 {
        field1: 2,
        field2: 4,
    }
    .get_a();
}
