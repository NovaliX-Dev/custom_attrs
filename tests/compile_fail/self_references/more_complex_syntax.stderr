error[E0106]: missing lifetime specifier
  --> tests/compile_fail/self_references/more_complex_syntax.rs:12:15
   |
12 |         list: &[u32],
   |               ^ expected named lifetime parameter
   |
help: consider introducing a named lifetime parameter
   |
5  ~ enum Enum<'a> {
6  |     #[attr(a = #self.5)]
 ...
11 |         #[allow(unused)]
12 ~         list: &'a [u32],
   |

error[E0425]: cannot find value `_5` in this scope
 --> tests/compile_fail/self_references/more_complex_syntax.rs:6:22
  |
6 |     #[attr(a = #self.5)]
  |                      ^ help: a local variable with a similar name exists: `_0`

error[E0425]: cannot find value `unknown` in this scope
 --> tests/compile_fail/self_references/more_complex_syntax.rs:9:22
  |
9 |     #[attr(a = #self.unknown)]
  |                      ^^^^^^^ not found in this scope

error[E0559]: variant `Enum::Variant2` has no field named `field1`
  --> tests/compile_fail/self_references/more_complex_syntax.rs:22:9
   |
22 |         field1: 2,
   |         ^^^^^^ `Enum::Variant2` does not have this field
   |
   = note: available fields are: `list`, `field2`
